.github/workflows/windows.yml

```diff
@@ -3,12 +3,37 @@ name: windows-latest
 on: [push]
 
 jobs:
-  do-everything: # name of the job. Jobs run in parallel unless specified otherwise.
+  compile: # name of the job. Jobs run in parallel unless specified otherwise.
+    name: compile
     runs-on: windows-latest
     strategy:
       matrix:
-        python-version: [ '3.6', '3.7' ]
-    name: Python ${{ matrix.python-version }}
+        python-version: ['3.6']
+        build-version: ["release", "debug"]
+    steps: # each - is a new sequentially run step
+      - uses: actions/checkout@master
+      - name: Setup python
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+          architecture: x64
+      - name: compile
+        shell: cmd
+        run: |
+          "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Enterprise\\VC\\Auxiliary\\Build\\vcvarsall.bat" x64 & python build.py ${{ matrix.build-version }}
+      - name: upload-planner
+        uses: actions/upload-artifact@master
+        with:
+          name: ${{ matrix.build-version }}
+          path: builds/${{ matrix.build-version }}
+
+  tests: # name of the job. Jobs run in parallel unless specified otherwise.
```

Comment:
I don't think we need separate jobs for compiling and testing (which includes uploading the compiled planner) because we only compile and test one version. I'd therefore move the steps of the test job into the compile job.

Reply:
You seem to call noLP/CPLEX/... builds different planner versions. And for each of those versions you build debug and release in the same job. I see everything with a different build command as new version (also debug and release).

The search test ("tox -e search") requires debug and release. Each compilation takes 15 minutes. Parallelizing it with the two jobs setup has the advantage that we get our feedback after 15 minutes, instead of 30. I think the configuration is still simple enough to keep the parallelization. The biggest disadvantage I see is that we setup 2 more virtual machines, but the setup times are quiete quick, thus, I guess it is not too expensive on the GitHub side.

Reply:
Another disadvantage is that if there is a compilation error in either debug or release, it most likely will also happen in the other build. Due to parallelization we will compile twice till the error, instead of only once.
As long as you push locally compiling code, this will happen rarely and I would ignore it, if the developer enables actions on his fork and regularly pushes incomplete stuff, it wastes some energy.